### PR TITLE
🦆

### LINKED ISSUE #33: toggle line number

### PR DESCRIPTION
**Pull request by elianiva**
- custom border background colour
- toggle line number (closes #33 )
- refactor validation
- refactor core logic
- refactor renderer

### CODE REVIEW

File: src/logic/validate.ts [hunk 1]
```diff
@@ -1,21 +1,66 @@
-import { BUNDLED_THEMES } from 'shiki';
-import type { ValidateFuncOptions } from '../types/function';
+import { BUNDLED_LANGUAGES, BUNDLED_THEMES } from 'shiki';
+import type { ValidOptions } from '../types/function';
+
+const validateBoolean = (name: string, bool: boolean, required: boolean): string => {
+  if (!bool && required) return `\`${name}\` is required!`;
+  if (bool && typeof bool !== 'boolean') return `${name} must be a boolean!`;
+  return '';
+};
```
Comment (aldy505) [inline src/logic/validate.ts:8]: dic, can you test this one out? line 6 kalo `bool` value nya false, kan dia bakal return error ya.

File: src/logic/svgRenderer.ts [hunk 2]
```diff
@@ -10,22 +10,67 @@ import { FontStyle, IThemedToken } from 'shiki';
 import type { HTMLEscapes, RendererOptions, SVGAttributes, SVGOutput } from '../types/renderer';
 
 const DEFAULT_CONFIG: Partial<RendererOptions> = {
-  fontFamily: 'JetBrainsMono Nerd Font',
+  fontFamily: 'JetBrains Mono',
```
Comment (aldy505) [inline src/logic/svgRenderer.ts:13]: ini kenapa diubah ke non-nerd font?

I mean it's ok but, ga mau sekalian ganti di `src/logic/genFont.ts` ? terus harus tuker yang ada di folder fonts

**Review by aldy505 — CHANGES_REQUESTED** [review]

File: src/logic/validate.ts (hunk 1 above)
Comment (elianiva) [reply src/logic/validate.ts:8]: oiya

File: src/logic/svgRenderer.ts (hunk 2 above)
Comment (elianiva) [reply src/logic/svgRenderer.ts:13]: sek, emang yg di folder itu versi nerd font? aku ganti soalnya ini sisaan pake font system, punyaku nerdfont

Comment (aldy505) [reply src/logic/svgRenderer.ts:13]: well of course. we both use starship arent we lmao

**Review by aldy505 — APPROVED** [review]

**Pull request merged**